client/modules/state/state.go

```diff
@@ -181,24 +133,18 @@ func (s *LevelDBState) Delete(key string) error {
 }
 
 func (s *LevelDBState) SaveOffset(offset uint64) error {
-	s.Lock()
-	defer s.Unlock()
-
 	bz := make([]byte, 8)
 	binary.LittleEndian.PutUint64(bz, offset)
 
-	if err := s.stateDb.Put(makeCompositeKey(s.topic, OffsetKey), bz, nil); err != nil {
+	if err := s.stateDb.Put(MakeCompositeKey(s.topic, OffsetKey), bz, nil); err != nil {
 		return fmt.Errorf("failed to set offset: %w", err)
 	}
 
 	return nil
 }
 
 func (s *LevelDBState) LoadOffset() (uint64, error) {
-	s.Lock()
```

Comment:
don't we need a mutex lock here?

Reply:
I think no, because func (*leveldb.DB).Get has a mutex inside